### PR TITLE
Refactor kindle content script with React

### PR DESCRIPTION
**Pull request by p-mazhnik**
Closes https://github.com/motiko/kcr-translate-ext/issues/21

- Tesseract moved to backend
- Listeners are handled properly

### CODE REVIEW

**Comment by motiko** [comment]
I'm not sure how I feel about injecting React into the page. It does increase page size significantly but the benefit is not very clear. WDYT?

**Comment by motiko** [comment]
> I'm not sure how I feel about injecting React into the page. It does increase page size significantly but the benefit is not very clear. WDYT? If the purpose it to unloadEvents I feel there are cheaper (performance wise) ways to achieve this. You want to have a call to discuss?

Seems like react size compared to overall resources is not very significant.
But as discussed let's try to keep reliance on it minimal in case we do want to remove it in the future. 
Right now lazy loading tesseract would bring much more benefit for faster loading :) .

File: chrome/background/background.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    // chrome.tabs.query({active: true, currentWindow: true}, function(tabs){
+    //   chrome.tabs.sendMessage(tabs[0].id, { command: "SETTINGS_UPDATED" }, function(response) {
+    //     if (!chrome.runtime.lastError) {
+    //       // if you have any response
+    //     } else {
+    //       // if you don't have any response it's ok but you should actually handle
+    //       // it and we are doing this when we are examining chrome.runtime.lastError
+    //     }
+    //   });
+    // });
```
Comment (p-mazhnik) [inline chrome/background/background.ts:74]: I want to go away from script reloading on settings changes in the next PR

**Comment by p-mazhnik** [comment]
By the way, @motiko `react_devtools_backend.js` file that we saw in the Chrome `Network` tab is actually a `React Developer Tools` extension's file and is not related to the current extension

File: chrome/background/background.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    setTimeout(function () {
+      sendResponse(true);
+    }, 1);
```
Comment (motiko) [inline chrome/background/background.ts:77]: What is the logic behind this timeout time?
If you want to run it in next tick you can use `0`

**Comment by motiko** [comment]
@p-mazhnik all in all looks good, left few small comments but can merge as is and fix later if you prefer. Let me know.

File: chrome/services/messaging.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  | IStartRecognitionMessage
+  | ISetProgressMessage;
+
+const getActiveTab = (winId?: number): Promise<chrome.tabs.Tab> => {
+  const config = { active: true, currentWindow: true };
+  if (winId) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    config.windowId = winId;
+  }
```
[the remark this answers is not in the document]
Comment (p-mazhnik) [reply chrome/services/messaging.ts]: Removed this function :)

File: chrome/background/background.ts (hunk 2 above)
Comment (p-mazhnik) [reply chrome/background/background.ts]: Oh, yeah it's just a mistake

File: chrome/services/messaging.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (motiko) [reply chrome/services/messaging.ts]: best code is no code 🤣

**Review by motiko — APPROVED** [review]

**Pull request merged**